wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+    PartialOrd,
+    Ord,
+    Clone,
+    Encode,
+    Decode,
+    Hash,
+    Debug,
+    TypeInfo,
+)]
+pub enum TradeableKittyError {
+    /// Error in the underlying money piece.
+    MoneyError(money::ConstraintCheckerError),
+    /// Error in the underlying kitty piece.
+    KittyError(kitties::ConstraintCheckerError),
+    /// Dynamic typing issue.
+    /// This error doesn't discriminate between badly typed inputs and outputs.
+    BadlyTyped,
+    /// Input missing for the transaction.
+    InputMissingError,
+    /// Not enough amount to buy a kitty.
+    InsufficientCollateralToBuyKitty,
+    /// The number of input vs number of output doesn't match for a transaction.
+    NumberOfInputOutputMismatch,
+    /// Can't buy more than one kitty at a time.
+    CannotBuyMoreThanOneKittyAtTime,
+    /// Kitty basic properties such as DNA, free breeding, and a number of breedings, are altered error.
+    KittyBasicPropertiesAltered,
+    /// Kitty not available for sale.Occur when price is None.
+    KittyNotForSale,
+    /// Kitty price cant be none when it is available for sale.
+    KittyPriceCantBeNone,
+    /// Duplicate kitty foundi.e based on the DNA.
+    DuplicateKittyFound,
+    /// Duplicate tradable kitty foundi.e based on the DNA.
+    DuplicateTradableKittyFound,
+    /// Kitty price is unaltered is not allowed for kitty price update transactions.
+    KittyPriceUnaltered,
+}
+
+impl From<money::ConstraintCheckerError> for TradeableKittyError {
+    fn from(error: money::ConstraintCheckerError) -> Self {
+        TradeableKittyError::MoneyError(error)
+    }
+}
+
+impl From<kitties::ConstraintCheckerError> for TradeableKittyError {
+    fn from(error: kitties::ConstraintCheckerError) -> Self {
+        TradeableKittyError::KittyError(error)
+    }
+}
+
+/// The main constraint checker for the trdable kitty piece. Allows below :
+/// Listing kitty for sale : multiple kiities are allowed, provided input and output in same order.
+/// Delisting kitty from sale : multiple tradable kiities are allowed, provided input and output in same order.
+/// Update kitty price : multiple tradable kiities are allowed, provided input and output in same order.
+/// Update kitty name : multiple tradable kiities are allowed, provided input and output in same order.
+/// Buy tradable kitty :  multiple tradable kiities are not allowed, Only single kiity operation is allowed.
+#[derive(
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Clone,
+    Encode,
+    Decode,
+    Hash,
+    Debug,
+    TypeInfo,
+)]
+pub enum TradableKittyConstraintChecker<const ID: u8> {
+    /// List kitty for sale, means kitty will converted to tradable kitty once transaction is executed
+    ListKittyForSale,
+    /// Delist kitty from sale, means tradable kitty will converted back to kitty  
+    DelistKittyFromSale,
+    /// Update price of tradable kitty.
+    UpdateKittyPrice,
+    // Update name of kitty
+    UpdateKittyName,
```

Comment:
Just my opinion, not a strong request for change. I think that updating the name should not be allowed here. If you want to change the name, there is already a path which is to de-list it, change its name, and re-list it.

A user might not want to click three different buttons to do this, but in that case it is the job of their UI to construct the three transactions for them.

I'll leave it up to you whether you want to take this suggestion or not. I don't feel strongly, just sharing the idea.

Reply:
Yes, its leads to 3 transactions to rename a Tradable kitty if I remove this. So please allow me to keep it.